### ISSUE TITLE
Revamping Syft to be light weight.

### ISSUE THREAD

**Issue by rasswanth-s** [OpenMined/PySyft#6959]
# # Problem.
Currently when we launch Syft in development mode, by the screenshot below it takes around 3.1 GB of RAM for a single node(might vary on other OS/machines).
![Screenshot 2022-11-04 at 10 44 13 AM](https://user-images.githubusercontent.com/43314053/199895557-885cdefa-3749-4809-90e1-90d2ae9c20c2.png)

This development mode is the worst case Scenario where we reinstall syft with additional packages in docker.

### Possible Solutions

- We are dependent on several third party libraries which are specified in setup.cfg , we could try removing the dependence on multiple libraries (Ex: currently we use rich for progress bar animation, in some places we also use tqdm,we could remove tqdm and replace entirely by rich).
- The frontend container takes 0.5 GB of memory , we could investigate on alternate packages for the same.
- When we launch a domain in development mode, we also install syft in development mode, Can we bypass the same?When we install syft in development mode, we install all the testing tools, which are not required in a domain node.

Making syft itself lightweight , would have a domino effect reducing memory footprint of `backend`, `celeryworker`,`backend-stream`.

# Expected Behaviour
To be able to launch atleast one Node on  a `low  RAM  and compute machines` (RAM<=8GB).

**Comment by shubham3121** [comment]
This could be because of the external packages we import when installing syft in these containers. e.g. pandas, torch, jax, etc. 
This ticket found libraries which cause the import of syft to be slow: https://github.com/OpenMined/PySyft/issues/6744, and I think lazyloading of these libraries will be essential at some point.

**Comment by IonesioJunior** [comment]
**Another HUGE advantage for these changes**:  the `Network` node uses `PySyft` for serializing and deserializing messages. But we don't use any Data Science package there at all _(which are by far the heaviest dependencies that we have)_. Unlocking the ability to install syft[headless] (without torch, numpy, scipy, etc) will` decrease the size of network images` and speed their `building/launching` process significantly.

**Comment by madhavajay** [comment]
> **Another HUGE advantage for these changes**: the `Network` node uses `PySyft` for serializing and deserializing messages. But we don't use any Data Science package there at all _(which are by far the heaviest dependencies that we have)_. Unlocking the ability to install syft[headless] (without torch, numpy, scipy, etc) will` decrease the size of network images` and speed their `building/launching` process significantly.

Yes, although currently when we receive certain types of messages we do need to deserialize them before we route them. If we can get a stable header format we can inspect and route messages without deserializing them which means we might be able to remove all the DL libraries or at least make them optional.
